jraft-rheakv/rheakv-core/src/main/java/com/alipay/sofa/jraft/rhea/client/failover/impl/FailoverClosureImpl.java

```diff
@@ -91,7 +92,12 @@ public void success(final T result) {
 
     @Override
     public void failure(final Throwable cause) {
-        this.future.completeExceptionally(cause);
+        if (cause instanceof RemotingException) {
```

Comment:
We have to consider the number of retries and only retry if `this.retriesLeft > 0`, otherwise it may result in an infinite loop.